packages/app/src/cli/services/app/config/push.ts

```diff
@@ -70,6 +75,7 @@ const getMutationVars = (app: App, configuration: CurrentAppConfiguration) => {
       : app.appProxy ?? undefined,
     posEmbedded: configuration.pos?.embedded ?? app.posEmbedded,
     preferencesUrl: configuration.app_preferences?.url ?? app.preferencesUrl,
+    requested_access_scopes: usesLegacyScopesBehavior(configuration) ? undefined : configuration.scopes ?? [],
```

Comment:
legacy or opted-out apps don't have their scopes pushed. all other apps have their scopes pushed, with a default of []